api/src/queries/configurations.js

```diff
@@ -1,88 +1,82 @@
-import { crudQueries } from 'co-postgres-queries';
+import omit from "lodash.omit";
+import client, {
+  insertOne as insertOneQuery,
+  updateOne as updateOneQuery
+} from "./knex";
 
-import db from './db';
-
-const table = 'configuration';
+const table = "configuration";
 const fields = [
-    'id',
-    'environment_id',
-    'name',
-    'state',
-    'default_format',
-    'created_at',
-    'updated_at',
+  "configuration.id",
+  "configuration.name",
+  "configuration.state",
+  "configuration.environment_id",
+  "configuration.default_format",
+  "configuration.created_at",
+  "configuration.updated_at",
+  "environment.name as environment_name",
+  "environment.project_id as project_id"
 ];
-const idFields = ['id'];
-const returnFields = ['id', 'name', 'state', 'default_format'];
-
-const query = crudQueries(table, fields, idFields, returnFields);
-
-query.selectPage = query.selectPage
-    .table(`
-        project
-        LEFT JOIN environment ON (project.id = environment.project_id)
-        LEFT JOIN configuration ON (environment.id = configuration.environment_id)
-    `)
-    .searchableFields(['project_id', 'environment.name', 'configuration.name'])
-    .idFields(['configuration.id'])
-    .returnFields([
-        'configuration.id',
-        'configuration.environment_id',
-        'configuration.name',
-        'configuration.state',
-        'configuration.default_format',
-        'configuration.created_at',
-        'configuration.updated_at',
-        'environment.name as environment_name',
-        'project.id as project_id',
-    ]);
 
 const findOne = async (projectId, environmentName, configurationName) => {
-    const client = await db.link(query);
-    const result = await client.selectPage(undefined, undefined, {
-        project_id: projectId,
-        'environment.name': environmentName,
-        'configuration.name': configurationName,
-        state: 'live',
+  const results = await client
+    .select(fields)
+    .from(table)
+    .innerJoin("environment", "environment.id", "configuration.environment_id")
+    .where({
+      project_id: projectId,
+      "environment.name": environmentName,
+      "configuration.name": configurationName,
+      "configuration.state": "live"
     });
-    client.release();
 
-    if (!result.length) {
-        return null;
-    }
-    return result[0];
+  return results[0];
 };
 
-const findAllByEnvironmentName = async (projectId, environmentName) => {
-    const client = await db.link(query);
-    const result = await client.selectPage(undefined, undefined, {
-        project_id: projectId,
-        'environment.name': environmentName,
-        state: 'live',
+const findAllByEnvironmentName = async (projectId, environmentName) =>
+  client
```

Comment:
add await?

Reply:
await is implicit (and lazy) if you return a promise :)